cgosqlite/cgosqlite.go

```diff
@@ -172,19 +184,21 @@ func (stmt *Stmt) Step() (row bool, err error) {
 	}
 }
 
-func (stmt *Stmt) StepResult() (row bool, lastInsertRowID, changes int64, err error) {
+func (stmt *Stmt) StepResult() (row bool, lastInsertRowID, changes int64, d time.Duration, err error) {
 	var rowid, chng C.sqlite3_int64
-	res := C.step_result(stmt.stmt, &rowid, &chng)
+	var duration C.int64_t
+	res := C.step_result(stmt.stmt, &rowid, &chng, &duration)
```

Comment:
Ditto: I think duration will be heap-allocated here.

One option here if you don't want to switch to returning a giant C struct is to heap allocate a single larger type:

```go
x := new(struct{
  rowid, chng C.sqlite3_int64
  duration C.int64_t
})
res := C.step_result(stmt.stmt, &x.rowid, &x.chng, &x.duration)
```

But better would be to return a C struct. (All this conditional on tests/benchmarks of the allocs here showing that that my mental escape analysis is correct.)